Expensify/App
# Create `ReportActionsList` to organize and Isolate problem code in `ReportActionsView`

### Details

This is an effort to clean up the `ReportActionsView` and start getting things under control. For now, I've focused on isolating stuff that more or less makes sense and separating it out from things that are confusing e.g.

- Setting the "last read action" for a report
- The Message counter stuff added in the past year
- `componentDidUpdate()` logic that handles a "new action that is not from the current user" event
- The "new marker" logic
- Basically anything related to unreads I'm leaving in the `ReportActionsView`

**Changes that were made**

- Anything related to the list of actions has been moved into a `ReportActionsList`
- The copy selection subscriber thing is now in it's own module so we can forget it exists
- Some minor improvements to the logic in `componentDidUpdate()` to make the conditional logic there a little clearer

**Note:** No logic should be changed in this PR as it is just a reorganization of existing code.

### Fixed Issues
$ https://github.com/Expensify/App/issues/8829

### Tests

Check for general regressions around the chat views, chat switching etc.

1. Open a chat and verify there are no errors
2. Verify things related to chat and chat messages appear to be normal with no major regressions


### PR Review Checklist
<details>
<summary><h4>PR Reviewer Checklist</h4></summary>


</details>

### QA Steps
Check for general regressions around the chat views, chat switching etc.

1. Open a chat and verify there are no errors
2. Verify things related to chat and chat messages appear to be normal with no major regressions


### Screenshots
❌
